lib/documents.js

```diff
@@ -2538,8 +2538,14 @@ function queryAllDocumentsImpl(query, jobOptions) {
     serverTimestamp: null
   };
 
-  if(jobState.jobOptions.consistentSnapshot && typeof jobState.jobOptions.consistentSnapshot !== 'boolean'){
-    throw new Error('consistentSnapshot needs to be a boolean.');
+  if(jobState.jobOptions.consistentSnapshot){
+    if(typeof jobState.jobOptions.consistentSnapshot !== 'boolean' &&
+        (typeof jobState.jobOptions.consistentSnapshot !== 'object' || !jobState.jobOptions.consistentSnapshot.valueOf('Timestamp'))){
```

Comment:
I'd suggest:

    typeof jobState.jobOptions.consistentSnapshot !== 'boolean &&
    !(jobState.jobOptions.consistentSnapshot instanceof Boolean) &&
    !(jobState.jobOptions.consistentSnapshotinstanceof instanceof mlutil.Timestamp)

JavaScript has both primitive and object versions of booleans and numbers (similar to Java), and the constructor of the Timestamp object can be used to verify the type similar to:

https://github.com/marklogic/node-client-api/blob/master/lib/documents.js#L1885

The same comment applies to the other initial validation.